src/hotspot/share/cds/archiveBuilder.hpp

```diff
@@ -142,6 +142,10 @@ class ArchiveBuilder : public StackObj {
         _buffered_addr = nullptr;
       }
     }
+    SourceObjInfo(address src, address buf) {
```

Comment:
This doesn't initialize all the fields, and should use initialization syntax.  Could you make one of these constructors call a common one that zeros out the other fields?

Reply:
I cleaned up this constructor and added comments.